File: boardgametracker.client/src/models/ScoreSheet/ScoreSheetTemplate.ts
export interface ScoreSheetTemplate {
  id?: number;
  name: string;
  description: string;
  mode: 'rounds' | 'categories' | 'hybrid';
  minPlayers: number;
  maxPlayers: number;
  version: string;
  jsonDefinition: string;
  gameId?: number;
  isPublic: boolean;
  createdByUserId: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface ScoreSheetTemplateDefinition {
  fields: ScoreField[];
  rules?: ScoreRule[];
  sections?: ScoreSection[];
}

export interface ScoreField {
  id: string;
  name: string;
  type: 'number' | 'text' | 'checkbox';
  defaultValue?: number | string | boolean;
  minValue?: number;
  maxValue?: number;
  isRequired?: boolean;
  sectionId?: string;
}

export interface ScoreSection {
  id: string;
  name: string;
  order: number;
}

export interface ScoreRule {
  id: string;
  name: string;
  expression: string;
  targetFieldId: string;
}

File: BoardGameTracker.Common/Entities/ScoreSheetTemplate.cs
using BoardGameTracker.Common.Entities.Helpers;

namespace BoardGameTracker.Common.Entities;

public class ScoreSheetTemplate : HasId
{
    public string Name { get; set; }
    public string Description { get; set; }
    public string Mode { get; set; } // "rounds" | "categories" | "hybrid"
    public int MinPlayers { get; set; } = 1;
    public int MaxPlayers { get; set; } = 10;
    public string Version { get; set; } = "1.0";
    public string JsonDefinition { get; set; }
    public int? GameId { get; set; }
    public Game Game { get; set; }
    public bool IsPublic { get; set; } = false;
    public string CreatedByUserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}